packages/eslint-plugin-svelte/src/utils/svelte-context.ts

```diff
@@ -0,0 +1,259 @@
+import type { RuleContext } from '../types.js';
+import fs from 'fs';
+import path from 'path';
+import { getPackageJsons } from './get-package-json.js';
+import { getFilename, getSourceCode } from './compat.js';
+
+const isRunInBrowser = !fs.readFileSync;
+
+export type SvelteContext = (
+	| ({
+			svelteVersion: '3/4';
+	  } & {
+			svelteFileType: '.svelte' | null;
+			runes: null;
+	  })
+	| ({
+			svelteVersion: '5';
+	  } & (
+			| {
+					svelteFileType: '.svelte' | '.svelte.[js|ts]';
+					/** If a user uses a parser other than `svelte-eslint-parser`, `undetermined` will be set. */
+					runes: boolean | 'undetermined';
```

Comment:
Hmm, I'm having a hard time understanding the difference between `null` and `'undetermined'` here - they both mean we couldn't figure it out, no?

Reply:
We can automatically recognize this via typing.

- null: The concept of runes itself does not exist.
- true: Using runes
- false: Not using runes
- undetermined: The concept of runes exists but I don't know if it is used

Reply:
Ah, ok. Is there any menaingful difference between `runes don't exist` and `not using runes`? Because they semm to me to be equivalent in all useful aspects...

Reply:
I think only facts should be expressed here, as usefulness depends on context